### PR TITLE
fix(releases) - Only update author name when it changes

### PR DESCRIPTION
**Pull request by wmak**
- Switch from `create_or_update` to `get_or_create`, since
  `create_or_update` will run an update everytime even if the data is
  the same.
- With these changes we should only update when the name has changed.

### CODE REVIEW

File: src/sentry/models/release.py [hunk 1]
```diff
@@ -403,15 +403,14 @@ def set_commits(self, commit_list):
                         author = None
                     elif author_email not in authors:
```
Comment (MeredithAnya) [inline src/sentry/models/release.py]: not that this is in your change and also not sure if I'm missing something but isn't `authors` always `{}` at this point? wondering if we need this elif/else

**Comment by MeredithAnya** [comment]
also https://github.com/getsentry/sentry/pull/11014

File: src/sentry/models/release.py [hunk 2]
```diff
@@ -403,15 +403,14 @@ def set_commits(self, commit_list):
...
                             email=author_email,
-                            values=author_data,
+                            defaults=author_data,
                         )
-                        if not created:
-                            author = CommitAuthor.objects.get(
-                                organization_id=self.organization_id, email=author_email
-                            )
+                        if author.name != author_data["name"]:
+                            author.name = author_data["name"]
```
Comment (wedamija) [inline src/sentry/models/release.py]: Probably better to just use `author.update(name=author_data["name"])`

**Review by wedamija — APPROVED** [review]

**Comment by wmak** [comment]
@MeredithAnya I think the integrity errors are from the `unique_together`s on the model ( at least that was the only way I could cause one ), and cause this get/create/update call doesn't touch `external_id` I don't think we can cause one here.

**Pull request merged**